### ISSUE TITLE
HMM Training  fails to converge

### ISSUE THREAD

**Issue by wichtounet** [mlpack/mlpack#538]
Hi, 

I'm trying to use mlpack to train HMM for word models, but most of my HMMs failed to converge with the given error:

> error: chol(): failed to converge

Here is the code (unfortunately, not standalone, it's inside another program): 
```cpp
    using GMM = mlpack::gmm::GMM;

    template<typename Distribution>
    using HMM = mlpack::hmm::HMM<Distribution>;

    using hmm_p = std::unique_ptr<HMM<GMM>>;

    constexpr const std::size_t n_gaussians = 1;

    const auto n_states_per_char = 1;
    const auto n_states = characters * n_states_per_char;
    const auto n_features = 9;

    auto hmm = std::make_unique<HMM<GMM>>(n_states, GMM(n_gaussians, n_features));

    std::vector<arma::mat> images(ref_a.size());
    std::vector<arma::Row<size_t>> labels(ref_a.size());

    //Fill images and labels...

     //Each image is initialized like this: 
     images[image] = arma::mat(n_features, width);
     labels[image] = arma::Row<size_t>(width);

    hmm->Train(images, labels);
```

I can give you the complete code, but it won't execute on its own. 

Looking at the features and labels, it seems OK, but I'm not a specialist in HMM at all. 

Is there some parameters I can play with to avoid this ? Is there anyway I can have some more verbose output that could help me investigate the issue ?

**Comment by rcurtin** [comment]
What happens if you vary `n_gaussians`?  Does it fail when set to 1?

Also, is there any more output?  If you compile with debugging symbols that may help.

This may be related to #496, but that issue should have been solved with the mlpack 2.0.1 release.

**Comment by wichtounet** [comment]
I tried setting different values for n_gaussians, but it does not make any difference. 

There are no more output even with all debug symbols. 

I have the 2.0.0, I haven't tried the 2.0.1, I'll do that now. You should really set the 2.0.1 as the "Latest release", right now it is a bit misleading since 2.0.0 has a huge "Latest release" green tag and a wall of text while 2.0.1 has no information and is almost not visible.

**Comment by wichtounet** [comment]
You were right, it was indeed fixed by the release 2.0.1. Thanks.

**Comment by rcurtin** [comment]
You're right, I forgot the publish the release notes!  Sorry about that.

I'm glad the updated version fixed it... I was scared that all the debugging I did for #496 was not enough. :)

For debugging symbols, yeah, I meant the CMake option DEBUG: `cmake -DDEBUG=ON` or similar.  If you have future problems with mlpack, you can try compiling with debugging symbols to get some more output that might be helpful for figuring out what is going on.
